python/ray/_private/runtime_env/image_uri.py

```diff
@@ -61,19 +61,28 @@ def _modify_context_impl(
         "--userns=keep-id",
     ]
 
-    # The RAY_RAYLET_PID and RAY_JOB_ID environment variables are
-    # needed for the default worker.
-    container_command.append("--env")
-    container_command.append("RAY_RAYLET_PID=" + os.getenv("RAY_RAYLET_PID"))
+    # Environment variables to set in container
+    env_vars = dict()
+
+    # Propagate all host environment variables that have the prefix "RAY_"
+    # This should include RAY_RAYLET_PID
+    for env_var_name, env_var_value in os.environ.items():
+        if env_var_name.startswith("RAY_"):
+            env_vars[env_var_name] = env_var_value
+
+    # Support for runtime_env['env_vars']
+    env_vars.update(context.env_vars)
+
+    # Set environment variables
+    for env_var_name, env_var_value in env_vars.items():
+        container_command.append("--env")
+        container_command.append(f"{env_var_name}='{env_var_value}'")
+
+    # The RAY_JOB_ID environment variable is needed for the default worker.
```

Comment:
isn't it already supposed to be handled by above code?

Reply:
No because at the time this code is run, `RAY_JOB_ID` hasn't been set in the environment yet. It will be set when the command to start the worker is kicked off, so instead of using `--env RAY_JOB_ID=0100000`, we use `--env RAY_JOB_ID=$RAY_JOB_ID`

Reply:
can you comment this? after that, let's merge